### PR TITLE
Add eslint workflow

### PR DESCRIPTION
**Pull request by teddywilson**

### CODE REVIEW

**Comment by milesalex** [comment]
@teddywilson What about markdown linting? Is that gonna be in another PR?

**Comment by teddywilson** [comment]
Yeah, let's start with js for now???

File: js/main.js [hunk 1]
```diff
@@ -1,51 +1,57 @@
...
+      const {state} = item.dataset;
       if (states[state] === undefined) {
         states[state] = [];
       }
       states[state].push(item);
     }
-    for (let [name, items] of Object.entries(states)) {
-      let stateElement = $("<div class='state'></div>");
+    for (const [name, items] of Object.entries(states)) {
+      const stateElement = $('<div class=\'state\'></div>');
```
Comment (milesalex) [inline js/main.js:20]:
```suggestion
      const stateElement = $(`<div class='state'></div>`);
```

**Comment by teddywilson** [comment]
ah shoot. I'll address in a later PR. I want to clean this file up anyway.

**Pull request merged**